app/controllers/tasks_controller.rb

```diff
@@ -60,4 +62,24 @@ def task_params
   def dashboard_path
     current_user.mentee? ? dashboard_mentee_profiles_path(week: @task.week) : dashboard_mentor_profiles_path(week: @task.week)
   end
+
+  def award_novice_badge
+      if @task.project.tasks.finished(current_user).count >= 1
```

Comment:
what does this line do?

Reply:
not sure if I remember correctly but mentor can also finish task? If yes,  we also need to make sure it's mentee that completed the task as only then we should add badge

Reply:
The above line counts if the no of tasks are are greater than 0. Yes we can do like that too. 

2) For mentor the case was like this:
The task can be added by either of them,once completed, the task can be marked finished by the one who created it. So either way, it denotes the task is completed by mentee. Hence it should be awarded badge.

Reply:
@pg20 don't we need to check that mentee is the one that completed the task then? if mentor can also do it if first task is completed by mentor, mentee will be rewarded badge anyway which is incorrect behavior.

You can see that in dashboard path we redirect to either mentee or mentor profiles - you can move this logic there if you don't want to check here whether current user is mentee

Reply:
@DMips In either case of the first task being completed by mentor or mentee (irrespective of who created it), wouldn't it be justified to assume it was a task for the mentee?
So, even if mentor marked it complete (perhaps after reviewing or something), the point is the first task of the project roadmap got completed and should be good to go for badge assignment. Is that not so?

Anyways, incase of task created by mentor is finished by mentee, the mentee will know it's still in review.

Reply:
@anjali-sharma I guess we can do that

Reply:
👍